### PR TITLE
CFE-3082: cf-check code cleanup, better errors, man page and other options (3.12)

### PR DESCRIPTION
**Pull request by olehermanse**

### CODE REVIEW

**Comment by olehermanse** [comment]
@cf-bottom jenkins

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/2451/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/2451/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-2451/

File: cf-check/backup.c [hunk 1]
```diff
@@ -106,6 +106,11 @@ int backup_files(Seq *filenames)
 int backup_main(int argc, const char *const *const argv)
 {
     Seq *files = argv_to_lmdb_files(argc, argv);
+    if (files == NULL || SeqLength(files) == 0)
+    {
+        Log(LOG_LEVEL_ERR, "No database files to back up");
+        return -1;
```
Comment (vpodzime) [inline cf-check/backup.c]: As mentioned in Slack, I realized this `-1` gets propagated as an exit code of the process and that's not good, exit codes should be non-negative.

Comment (olehermanse) [reply cf-check/backup.c]: Thanks! I usually cherry-pick whole PRs so they have the matching title. I added the cherry-pick label to the follow-up PR, so I wont forget. https://github.com/cfengine/core/pull/3769

**Pull request merged**